onset_maproom/tests/test_calc.py

```diff
@@ -3,11 +3,144 @@
 import xarray as xr
 
 import calc
+import data_test_calc
 
-def test_onset():
+
+def test_daily_tobegroupedby_season_cuts_on_days():
+
+    precip = data_test_calc.multi_year_data_sample()
+    dts = calc.daily_tobegroupedby_season(precip, 29, 11, 29, 2)
+
+    assert dts["T"].size == 461
+
+
+def test_daily_tobegroupedby_season_creates_groups():
+
+    precip = data_test_calc.multi_year_data_sample()
+    dts = calc.daily_tobegroupedby_season(precip, 29, 11, 29, 2)
+
+    assert dts["group"].size == 5
+
+
+def test_daily_tobegroupedby_season_picks_right_end_dates():
+
+    precip = data_test_calc.multi_year_data_sample()
+    dts = calc.daily_tobegroupedby_season(precip, 29, 11, 29, 2)
+    assert (
+        dts.seasons_ends
+        == pd.to_datetime(
+            xr.DataArray(
+                [
+                    "2001-02-28T00:00:00.000000000",
+                    "2002-02-28T00:00:00.000000000",
+                    "2003-02-28T00:00:00.000000000",
+                    "2004-02-29T00:00:00.000000000",
+                    "2005-02-28T00:00:00.000000000",
+                ],
+                dims=["group"],
+                coords={"group": dts.seasons_ends["group"]},
+            )
+        )
+    ).all()
+
+
+def test_seasonal_onset_date_keeps_returning_same_outputs():
+
+    precip = data_test_calc.multi_year_data_sample()
+    onsetsds = calc.seasonal_onset_date(
+        daily_rain=precip,
+        search_start_day=1,
+        search_start_month=3,
+        search_days=90,
+        wet_thresh=1,
+        wet_spell_length=3,
+        wet_spell_thresh=20,
+        min_wet_days=1,
+        dry_spell_length=7,
+        dry_spell_search=21,
+        time_coord="T",
+    )
+    onsets = onsetsds.onset_delta + onsetsds["T"]
+    assert np.array_equal(
+        onsets,
+        pd.to_datetime(
+            xr.DataArray(
+                [
+                    "NaT",
+                    "2001-03-08T00:00:00.000000000",
+                    "NaT",
+                    "2003-04-12T00:00:00.000000000",
+                    "2004-04-04T00:00:00.000000000",
+                ],
+                dims=["T"],
+                coords={"T": onsets["T"]},
+            )
+        ),
```

Comment:
I don't see what purpose the DataArray serves here. I think you can construct the same pandas structure directly:
```
pd.to_datetime(
                [
                    "NaT",
                    "2001-03-08T00:00:00.000000000",
                    "NaT",
                    "2003-04-12T00:00:00.000000000",
                    "2004-04-04T00:00:00.000000000",
                ],
)

Reply:
That is mostly driven by the fact that it's unclear to me what the == is testing. Just the values? or also the structure/nature of the array? and if so to what extent?
It does work without the xarray attributes and so made the changes accordingly.
I'm going to merge now, if it lets me.

Reply:
`NaT == NaT` is false, so you can't use `==` to compare two arrays.

You replaced it with `np.array_equal`. Numpy doesn't support dimension names or coordinate values, so all this compares is the values of the data variable (onset date in this case). If you wanted to compare coordinate values as well, you could have used `pd.testing.assert_frame_equal` or `xr.testing.assert_equal`. According to the doc, `xr.testing.assert_equal` still ignores dimension names.